ONSdigital/dp-api-tests
# Feature/filter blueprint ouput

### What

Acceptance test rework for filter API 

Filter blueprint resource can be updated and submitted as many times as
the user wants. On submission a filter output resource is created and
will be updated with download urls by the backend export process. When
submitting a filter blueprint, the response body will return a link to
the filter output resource.

### How to review

Check logic makes sense and follows spec. Make sure all acceptance tests pass and do not leave data in mongo db. To successfully run filterAPI acceptance tests, do the following:

1) Make sure mongo db instance is running, can use mongod & (if this doesn;t work you probably need to install mongo)
2) git clone dp-dataset-api (`cmd-develop` branch will work)
3) Run dataset API (make debug)
4) git clone dp-filter-api
5) git checkout `feature/create-filter-output-resource`
6) Run filter API (make debug)
7) Run acceptance tests:
  a) cd filterAPI directory in dp-api-tests repo.
  b) go test ./...

### Who can review

Team A or B

Notes:

Pull request needs to go in with the following PR's:

FE services:
* go-ns (filter-api client): https://github.com/ONSdigital/go-ns/pull/38#pullrequestreview-74029213
* frontend-filter-dataset-controller: https://github.com/ONSdigital/dp-frontend-filter-dataset-controller/pull/37/files
* frontend-dataset-controller: https://github.com/ONSdigital/dp-frontend-dataset-controller/pull/19#pullrequestreview-74030419
* frontend-router: https://github.com/ONSdigital/dp-frontend-router/pull/47/files

BE services:
* dp-filter-api: https://github.com/ONSdigital/dp-filter-api/pull/38
* dp-dataset-exporter: https://github.com/ONSdigital/dp-dataset-exporter/pull/13
* dp-dataset-exporter-xlsx: ONSdigital/dp-dataset-exporter-xlsx#11
